script/release.js

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env node
+
+require('colors')
+const args = require('minimist')(process.argv.slice(2))
+const assert = require('assert')
+const fs = require('fs')
+const { execSync } = require('child_process')
+const GitHub = require('github')
+const { GitProcess } = require('dugite')
+const nugget = require('nugget')
+const pkg = require('../package.json')
+const pkgVersion = `v${pkg.version}`
+const pass = '\u2713'.green
+const path = require('path')
+const fail = '\u2717'.red
+const sumchecker = require('sumchecker')
+const temp = require('temp').track()
+const { URL } = require('url')
+let failureCount = 0
+
+assert(process.env.ELECTRON_GITHUB_TOKEN, 'ELECTRON_GITHUB_TOKEN not found in environment')
+
+const github = new GitHub()
+github.authenticate({type: 'token', token: process.env.ELECTRON_GITHUB_TOKEN})
+const gitDir = path.resolve(__dirname, '..')
+
+async function getDraftRelease (version) {
+  let releaseInfo = await github.repos.getReleases({owner: 'electron', repo: 'electron'})
+  let drafts
+  if (version) {
+    drafts = releaseInfo.data
+      .filter(release => release.tag_name === version)
+  } else {
+    drafts = releaseInfo.data
+      .filter(release => release.draft)
+  }
+  check(drafts.length === 1, 'one draft exists', true)
+  const draft = drafts[0]
+  check(draft.tag_name === `v${pkg.version}`, `draft release version matches local package.json (v${pkg.version})`)
+  check(draft.prerelease, 'draft is a prerelease')
+  check(draft.body.length > 50 && !draft.body.includes('(placeholder)'), 'draft has release notes')
+  return draft
+}
+
+async function validateReleaseAssets (draft) {
+  const requiredAssets = assetsForVersion(draft.tag_name).sort()
+  const extantAssets = draft.assets.map(asset => asset.name).sort()
+  const downloadUrls = draft.assets.map(asset => asset.browser_download_url).sort()
+
+  requiredAssets.forEach(asset => {
+    check(extantAssets.includes(asset), asset)
+  })
+  check((failureCount === 0), `All required GitHub assets exist for release`, true)
```

Comment:
In the case where one of the checks before this one has failed, but the process hasn't exited, you would get an inaccurate error message saying `All required GitHub assets exist for release` even if the assets are all there.